dts/client.py

```diff
@@ -104,47 +106,68 @@ def search(self,
                status = None,
                offset = 0,
                limit = None,
+               specific = None,
     ):
         """
 `client.search(database = None,
                query = None,
                status = None,
                offset = 0,
-               limit = None) -> `list` of `frictionless.DataResource` objects
+               limit = None,
+               specific = None) -> `list` of `frictionless.DataResource` objects
 
 * Performs a synchronous search of the database with the given name using the
   given query string.
 Optional arguments:
+    * query: a search string that is directly interpreted by the database
     * status: filters for files based on their status:
         * `"staged"` means "search only for files that are already in the source database staging area"
-        * `"archived"` means "search only for files that are archived and not staged"
+        * `"unstaged"` means "search only for files that are not staged"
     * offset: a 0-based index from which to start retrieving results (default: 0)
     * limit: if given, the maximum number of results to retrieve
+    * specific: a dictionary mapping database-specific search parameters to their values
 """
+        params = {
+            'database': database,
+            'query':    query,
+        }
         if not self.uri:
             raise RuntimeError('dts.Client: not connected.')
-        if type(database) != str:
+        if not isinstance(query, str):
+            raise RuntimeError('search: missing or invalid query.')
```

Comment:
is there any case in which we might want to search for an integer or number? e.g. if some DB used numeric IDs (like the JGI taxon_oids)? Or will this `query` already have been processed to include any extra stuff like the db field to search?

Reply:
Good question! I've just been assuming that a query is a string. Do you mind if we keep it this way for now and see what problems arise?

Reply:
the easiest approach would just be to convert it to a string if it is an int / float / boolean, so that those bases are covered